File: 7_kyu/Length_and_two_values.go
package kata

// Creates an array of n values.
// The values are alternatend first, second, first, ...
func Alternate(n int, firstValue string, secondValue string) []string {
  if n < 0 {
    // error case
    panic("argument: negative length")
  }
  arr := make([]string, 0, n)
  for remaining := n; remaining > 0; remaining-- {
    arr = append(arr, firstValue)
    remaining--
    if remaining > 0 { 
      arr = append(arr, secondValue)
    }
  }
  return arr
}

File: 7_kyu/Length_and_two_values.rs
// Creates vector of length n,
// filled alternated by first_value and second_value.
fn alternate<'a>(n: usize, first_value: &'a str, second_value: &'a str) -> Vec<&'a str> {
    let mut result = Vec::with_capacity(n);
    while result.len() < n {
        result.push(first_value);
        if result.len() < n {
            result.push(second_value);
        }
    }
    result
}
